api/v1beta1/module_webhook.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"regexp"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// log is for logging in this package.
+var modulelog = logf.Log.WithName("module-resource")
+
+func (r *Module) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/validate-kmm-sigs-x-k8s-io-v1beta1-module,mutating=false,failurePolicy=fail,sideEffects=None,groups=kmm.sigs.x-k8s.io,resources=modules,verbs=create;update,versions=v1beta1,name=vmodule.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Validator = &Module{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *Module) ValidateCreate() error {
+	modulelog.Info("validate create", "name", r.Name)
+
+	if err := r.validateKernelMappingRegex(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+func (r *Module) ValidateUpdate(old runtime.Object) error {
+	modulelog.Info("validate update", "name", r.Name)
+
+	if err := r.validateKernelMappingRegex(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
+func (r *Module) ValidateDelete() error {
+	return nil
+}
+
+func (r *Module) validateKernelMappingRegex() error {
+	if r.Spec.ModuleLoader.Container.KernelMappings == nil {
+		return nil
+	}
+
+	for _, km := range r.Spec.ModuleLoader.Container.KernelMappings {
+		if _, rErr := regexp.Compile(km.Regexp); rErr != nil {
+			return errors.New("invalid regex: " + rErr.Error())
+		}
+	}
```

Comment:
```suggestion
	for idx, km := range r.Spec.ModuleLoader.Container.KernelMappings {
		if km.Regexp == "" {
			continue
		}
	
		if _, rErr := regexp.Compile(km.Regexp); rErr != nil {
			return fmt.Errorf("invalid regex at index %d: %v", idx, err)
		}
	}
```

Reply:
An empty regexp is valid and will pass the `Compile` check, should we split that and do redundant checks?

I'll add the index to explain better the error to the user :+1: